src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -1295,6 +1351,114 @@ public int GetNumber(int numberIndex)
+                    int extendedNumberCount = ReadInt16(data, extendedBeginning + 2);
+                    int extendedStringCount = ReadInt16(data, extendedBeginning + 4);
+                    int extendedStringNumOffsets = ReadInt16(data, extendedBeginning + 6);
+                    int extendedStringTableByteSize = ReadInt16(data, extendedBeginning + 8);
+                    if (extendedBoolCount < 0 ||
+                        extendedNumberCount < 0 ||
+                        extendedStringCount < 0 ||
+                        extendedStringNumOffsets < 0 ||
+                        extendedStringTableByteSize < 0)
+                    {
+                        // The extended header contained invalid data.  Bail.
+                        return extendedStrings;
+                    }
+
+                    // Skip over the extended bools.  We don't need them now and can add this in later 
+                    // if needed. Also skip over extended numbers, for the same reason.
+
+                    // Get the location where the extended string offsets begin.  These point into
+                    // the extended string table.
+                    int extendedOffsetsStart =
+                        extendedBeginning + // go past the normal data
+                        ExtendedHeaderSize + // and past the extended header
+                        RoundUpToEven(extendedBoolCount) + // and past all of the extended Booleans
+                        (extendedNumberCount * 2); // and past all of the extended numbers
+
+                    // Get the location where the extended string table begins.  This area contains
+                    // null-terminated strings.
+                    int extendedStringTableStart =
+                        extendedOffsetsStart +
+                        (extendedStringCount * 2) + // and past all of the string offsets
+                        ((extendedBoolCount + extendedNumberCount + extendedStringCount) * 2); // and past all of the name offsets
+
+                    // Get the location where the extended string table ends.  We shouldn't read past this.
+                    int extendedStringTableEnd =
+                        extendedStringTableStart +
+                        extendedStringTableByteSize;
+
+                    if (extendedStringTableEnd > data.Length)
+                    {
+                        // We don't have enough data to parse everything.  Bail.
+                        return extendedStrings;
+                    }
+
+                    // Now we need to parse all of the extended string values.  These aren't necessarily
+                    // "in order", meaning the offsets aren't guaranteed to be increasing.  Instead, we parse
+                    // the offsets in order, pulling out each string it references and storing them into our
+                    // results list in the order of the offsets.
+                    var values = new List<string>();
```

Comment:
nit: Since we know how many values we expect there to be, can we initialize the list with that many elements? It may hurt if the data turns out to be malformed, but I'm assuming that is probably rare.